duck/duckdb.go

```diff
@@ -261,3 +270,45 @@ func resultsToFrame(name string, res string, f *sdk.Frame, frames []*sdk.Frame)
 // 	}
 // 	return nil
 // }
+
+func exclude(query string) (bool, string) {
+	query = strings.ToLower(query)
+	for _, exclusion := range exclusions {
+		if strings.Contains(query, exclusion) {
+			return true, exclusion
+		}
+	}
+	// check for "set" command
+	// can't just add to exclusions since that uses Contains
+	// select * from corset - should be ok
+	parts := strings.Split(query, " ")
+	for _, part := range parts {
+		if strings.TrimSpace(part) == "set" {
+			return true, "set"
+		}
+	}
+	return false, ""
+}
+
+var exclusions = []string{
+	".cd",
+	".import",
+	".log",
+	".mode",
+	".once",
+	".open",
+	".output",
+	".parameter",
+	".read",
+	".shell",
+	".show",
+	".system",
+	".testcase",
+	".timer",
```

Comment:
How did you build this list? Wouldn't an allow-list be safer? What if future versions of DuckDB ship with more dot-commands, or something like that.

Reply:
all good questions.  i went through all the commands to determine what could be malicious.  if we update the duckdb version in grafana cloud, we need to revisit this.  but .  again, this is just a quick fix.  you could make the argument that the sql validation should really happen in grafana.  some users of this lib might want to use these commands.  these exclusions only happen when querying from a frame ( which probably only happens from grafana )